spark/src/main/scala/org/apache/spark/sql/delta/commands/alterDeltaTableCommands.scala

```diff
@@ -322,7 +322,7 @@ case class AlterTableDropFeatureDeltaCommand(
     // Check whether the protocol contains the feature in either the writer features list or
     // the reader+writer features list. Note, protocol needs to denormalized to allow dropping
     // features from legacy protocols.
-    val protocol = table.initialSnapshot.protocol
+    val protocol = table.deltaLog.update().protocol
```

Comment:
I'm happy with changing this to `deltaLog.update()`, but just wondering if there was any risk before with using `initialSnapshot`. Don't know why we haven't ran into a problem with this before since we should always get the latest protocol, not the "The snapshot initially associated with this table. "